openwisp_controller/config/tests/test_selenium.py

```diff
@@ -130,6 +132,48 @@ def test_create_new_device(self):
             'The Device “11:22:33:44:55:66” was added successfully.',
         )
 
+    def test_device_preview_keyboard_shortcuts(self):
+        self._create_config(device=self._create_device(name='Test'))
+        self.login()
+        self.open(reverse('admin:config_device_changelist'))
+
+        # open first device in the list
+        self.web_driver.find_element(
+            by=By.CSS_SELECTOR, value='tbody tr:nth-child(1) th a'
+        ).click()
+        try:
+            WebDriverWait(self.web_driver, 2).until(
+                EC.presence_of_element_located((By.CSS_SELECTOR, '#content-main'))
+            )
+        except TimeoutException:
+            self.fail('Device detail page did not load in time')
```

Comment:
The `open` method already uses `WebDriverWait`. We can just wrap it in a try-except block.

https://github.com/openwisp/openwisp-utils/blob/1103df50164d7a6ac9b2cefa2f866f580b688111/openwisp_utils/test_selenium_mixins.py#L68-L70